cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler/ArgumentsHandlerTest.kt

```diff
@@ -0,0 +1,202 @@
+
+import de.fraunhofer.aisec.cpg.TranslationResult
+import de.fraunhofer.aisec.cpg.frontends.python.PythonLanguage
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.get
+import de.fraunhofer.aisec.cpg.test.analyze
+import java.nio.file.Path
+import kotlin.collections.component1
+import kotlin.collections.component2
+import kotlin.collections.iterator
+import kotlin.test.Test
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import org.junit.jupiter.api.BeforeAll
+
+class ArgumentsHandlerTest {
+
+    companion object {
+        private lateinit var topLevel: Path
+        private lateinit var result: TranslationResult
+
+        @JvmStatic
+        @BeforeAll
+        fun setup() {
+            topLevel = Path.of("src", "test", "resources", "python")
+            analyzeFile("arguments.py")
+        }
+
+        fun analyzeFile(fileName: String) {
+            result =
+                analyze(listOf(topLevel.resolve(fileName).toFile()), topLevel, true) {
+                    it.registerLanguage<PythonLanguage>()
+                }
+            assertNotNull(result)
+        }
+    }
+
+    @Test
+    fun testPosOnlyArguments() {
+        var func = result.functions["pos_only_and_args"]
+        assertNotNull(func)
+
+        val list = mapOf("a" to true, "b" to true, "c" to false)
+        list.keys.forEachIndexed { idx, name ->
+            var param = func.parameterEdges.firstOrNull { it.end.name.localName == name }
+            assertNotNull(param, "$name should not be empty")
+            if (list[name] == true) {
+                assertContains(
+                    param.end.modifiers,
+                    PythonLanguage.Companion.MODIFIER_POSITIONAL_ONLY_ARGUMENT
+                )
+            }
+            assertEquals(idx, param.index)
+        }
+    }
+
+    @Test
+    fun testVarargsArguments() {
+        var func = result.functions["test_varargs"]
+        assertNotNull(func, "Function 'test_varargs' should be found")
+
+        var variadicArg = func.parameters["args"]
+        assertNotNull(variadicArg, "Failed to find variadic args")
+        assertEquals(true, variadicArg.isVariadic)
+    }
+
+    @Test
+    fun testKwOnlyArguments() {
+        var func = result.functions["kwd_only_arg"]
+        assertNotNull(func, "Function 'kwd_only_arg' should be found")
+
+        var kwOnlyArg = func.parameters["arg"]
+        assertNotNull(kwOnlyArg, "Failed to find keyword only args")
+        assertContains(kwOnlyArg.modifiers, PythonLanguage.Companion.MODIFIER_KEYWORD_ONLY_ARGUMENT)
+    }
+
+    @Test
+    fun testKwDefaultArguments() {
+        var func = result.functions["kw_defaults"]
+        assertNotNull(func, "Function 'kw_defaults' should be found")
+
+        var kwOnlyParams = mapOf("c" to 2, "d" to null, "e" to 3)
+
+        for ((paramName, expectedDefaultValue) in kwOnlyParams) {
```

Comment:
Does it make sense to add a check that there are no additional arguments (e.g. by checking the size of `func.parameters` is as expected)?

Reply:
Done